mbed-trace/mbed_trace.h

```diff
@@ -273,34 +276,41 @@ char* mbed_trace_ipv6_prefix(const uint8_t *prefix, uint8_t prefix_len);
  */
 char* mbed_trace_array(const uint8_t* buf, uint16_t len);
 
-#else // defined(__GNUC__) || defined(__CC_ARM)
-int  mbed_trace_init( void );
-void mbed_trace_free( void );
-void mbed_trace_buffer_sizes(int lineLength, int tmpLength);
-void mbed_trace_config_set(uint8_t config);
-uint8_t mbed_trace_config_get(void);
-void mbed_trace_prefix_function_set( char* (*pref_f)(size_t) );
-void mbed_trace_suffix_function_set(char* (*suffix_f)(void) );
-void mbed_trace_print_function_set( void (*print_f)(const char*) );
-void mbed_trace_cmdprint_function_set( void (*printf)(const char*) );
-void mbed_trace_exclude_filters_set(char* filters);
-const char* mbed_trace_exclude_filters_get(void);
-void mbed_trace_include_filters_set(char* filters);
-const char* mbed_trace_include_filters_get(void);
-void mbed_tracef(uint8_t dlevel, const char* grp, const char *fmt, ...);
-const char* mbed_trace_last(void);
-char* mbed_trace_array(const uint8_t* buf, uint16_t len);
-#if YOTTA_CFG_MBED_TRACE_FEA_IPV6 == 1
-char* mbed_trace_ipv6(const void *addr_ptr);
-char* mbed_trace_ipv6_prefix(const uint8_t *prefix, uint8_t prefix_len);       
-#endif // YOTTA_CFG_MBED_TRACE_FEA_IPV6
-
-#endif // defined(__GNUC__) || defined(__CC_ARM)
+#ifdef __cplusplus
+}
+#endif
 
+#endif /* MBED_TRACE_H_ */
 
-#else // YOTTA_CFG_MBED_TRACE
+/* These macros are outside the inclusion guard so they will be re-evaluated for each individual compilation unit and
+ * if tracing is disabled, the dummies will hide the real functions. The real functions can still be reached by
+ * surrounding the name of the function with brackets, e.g. "(mbed_tracef)(dlevel, grp, "like so");"
+ * */
+#if defined(YOTTA_CFG_MBED_TRACE) || (defined(YOTTA_CFG) && !defined(NDEBUG))
```

Comment:
FEA_TRACE_SUPPORT going to go in via the separate commit, and then resolving the merge conflict?